ckanext/harvest/plugin/__init__.py

```diff
@@ -29,6 +29,8 @@ class DefaultTranslation():
 
 if p.toolkit.check_ckan_version(min_version='2.9.0'):
     from ckanext.harvest.plugin.flask_plugin import MixinPlugin
+elif p.toolkit.check_ckan_version(min_version='2.10.0'):
```

Comment:
We can't do that @TomeCirun, otherwise we will miss all the methods that `flask_plugin.MixinPlugin` implements. If we wanted to add another mixin we would need to import a separate mixin class on CKAN >= 2.10 and use it on the `Harvest` class:

```python
class Harvest(MixinPlugin, MixinPlugin2, p.SingletonPlugin, DefaultDatasetForm, DefaultTranslation):
```

But let's not do that. Let's just keep all the changes in this file and implement both hooks for CKAN pre and post 2.10. As the function signatures haven't changed we can just reference the new hooks from the old ones. Something like:

```python 
class Harvest(MixinPlugin, p.SingletonPlugin, DefaultDatasetForm, DefaultTranslation):
    # ...
    p.implements(p.IPackageController, inherit=True)

    # IPackageController

    # CKAN < 2.10 hooks
    def after_create(self, context, data_dict):
        self.after_dataset_create(context, data_dict)

    def after_update(self, context, data_dict):
        self.after_update_update(context, data_dict)
   # etc...

   # CKAN >= 2.10 hooks

   def after_dataset_create(self, context, data_dict):
       if 'type' in data_dict and data_dict['type'] == DATASET_TYPE_NAME and not self.startup:
            # Create an actual HarvestSource object
            _create_harvest_source_object(context, data_dict)

   def after_dataset_update(self, context, data_dict):
       if 'type' in data_dict and data_dict['type'] == DATASET_TYPE_NAME:
            # Edit the actual HarvestSource object
            _update_harvest_source_object(context, data_dict)

   # etc...

```
Does this make sense @TomeCirun ?

Reply:
Yep, that makes sense, thanks for your help @amercader
At first, I was thinking to make it in the same file and put comments for each, but then I ended up with a separate plugin and forgot about method implementations...